tests/sparseml/pytorch/sparsification/distillation/test_per_layer_distillation.py

```diff
@@ -0,0 +1,410 @@
+            fake_loss,
+            model,
+            optimizer,
+            modifier.start_epoch,
+            test_steps_per_epoch,
+            student_inputs=student_inputs,
+            student_outputs=student_outputs,
+        )
+
+        assert isinstance(updated_loss, torch.Tensor)
+        assert updated_loss.shape == fake_loss.shape
+        assert fake_loss.item() != updated_loss.item()
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+def test_distillation_modifier_yaml():
+    truth = PerLayerDistillationModifier(
+        gain=2.0,
+        start_epoch=0.0,
+        end_epoch=12.0,
+        update_frequency=1.5,
+        normalize=False,
+        student_layer_names=["a", "b", "c"],
+        teacher_layer_names=["d", "e", "f"],
+        project_features=False,
+        epsilon=3.0,
+    )
+    yaml_str = f"""
+        !PerLayerDistillationModifier
+            gain: {truth.gain}
+            start_epoch: {truth.start_epoch}
+            end_epoch: {truth.end_epoch}
+            update_frequency: {truth.update_frequency}
+            normalize: {truth.normalize}
+            student_layer_names: {truth.student_layer_names}
+            teacher_layer_names: {truth.teacher_layer_names}
+            project_features: {truth.project_features}
+            epsilon: {truth.epsilon}
+        """
+    from_yaml = PerLayerDistillationModifier.load_obj(yaml_str)
+    twice_from_yaml = PerLayerDistillationModifier.load_obj(str(from_yaml))
+
+    assert isinstance(from_yaml, PerLayerDistillationModifier)
+    for attr_name in [
+        "gain",
+        "start_epoch",
+        "end_epoch",
+        "update_frequency",
+        "normalize",
+        "student_layer_names",
+        "teacher_layer_names",
+        "project_features",
+        "epsilon",
+    ]:
+        assert (
+            getattr(truth, attr_name)
+            == getattr(from_yaml, attr_name)
+            == getattr(twice_from_yaml, attr_name)
+        )
+
+
+@pytest.mark.skipif(
+    os.getenv("NM_ML_SKIP_PYTORCH_TESTS", False),
+    reason="Skipping pytorch tests",
+)
+def test_optimizer_serialization_with_projections(tmp_path):
+    # since we are adding param groups to optimizer during initialization, we need
+    # to ensure that they can be serialized/reloaded properly
+    optim_checkpoint_path = tmp_path / "optim.pth"
+    student = mlp(12, 24, 64)
+    teacher = mlp(12, 24, 64)
+
+    modifier = PerLayerDistillationModifier()
+    modifier.initialize(module=student, distillation_teacher=teacher)
+
+    optimizer = create_optim_sgd(student)
+    x = torch.randn(5, 12)
+    fake_loss = student(x).mean()
+
+    # add state to optimizer
+    modifier.loss_update(
+        loss=fake_loss,
+        module=student,
+        optimizer=optimizer,
+        epoch=modifier.start_epoch,
+        steps_per_epoch=10,
+        student_inputs=x,
+        student_outputs=fake_loss,
+    )
+
```

Comment:
let's add some sanity checks that the projection param group exists

Reply:
Done!